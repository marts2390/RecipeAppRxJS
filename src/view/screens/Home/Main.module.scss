@import "src/view/styles/variables.scss";

.homeScreen {
  padding: 0.5em;
  max-width: 1600px;
  margin: 0 auto;

  @media (min-width: $screen-md) {
    padding: 2em;
  }

  h1 {
    font-size: $font-xl;
    font-weight: $font-medium;
    margin-top: 0;
  }

  .header {
    padding: 1.5em;
    background-color: $white;
    margin-bottom: 2em;
    border-radius: 10px;

    h4 {
      font-weight: $font-medium;
      margin: 0;
      margin-bottom: 1em;
    }
  }
}
