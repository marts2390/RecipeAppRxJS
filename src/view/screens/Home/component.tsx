import React, { FC } from 'react'
import { ICategoriesDataModel } from 'models/categoriesDataModel'

import Slider from 'view/components/Slider'
import RecipeList from './RecipeList'

import styles from './Main.module.scss'

interface IHomeComponentProps {
  categories: ICategoriesDataModel[]
  getRecipesByCategory: (category: string) => void;
}

const HomeComponent:FC<IHomeComponentProps> = ({ categories, getRecipesByCategory }) => (
  <div className={ styles.homeScreen }>
    <div className={ styles.header }>
      <h1>Choose Your Recipe!</h1>
      <Slider
        sliderItems={ categories }
        sliderAction={ getRecipesByCategory }
      />
    </div>
    <div className={ styles.body }>
      <RecipeList />
    </div>
  </div>
)

export default HomeComponent
