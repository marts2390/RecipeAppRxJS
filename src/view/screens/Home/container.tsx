import React, { FC } from 'react'
import { useObservable } from 'rxjs-hooks'
import UC from 'actions'

import HomeComponent from './component'

const HomeContainer:FC = () => {
  const { categoriesData } = useObservable(UC.HomeScreenService.getSubject) || UC.HomeScreenService.getCurrentState()

  return (
    <HomeComponent
      categories={ categoriesData }
      getRecipesByCategory={ UC.HomeScreenService.getRecipesByCategory }
    />
  )
}

export default HomeContainer
