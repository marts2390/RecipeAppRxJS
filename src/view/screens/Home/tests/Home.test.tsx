import React from 'react'
import { shallow } from 'enzyme'
import Home from '../index'

test('Home page is rendered', () => {
  const wrapper = shallow(<Home />)
  expect(wrapper).toMatchSnapshot()
})
