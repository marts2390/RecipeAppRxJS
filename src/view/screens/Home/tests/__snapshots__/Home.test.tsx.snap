// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page is rendered 1`] = `
<HomeComponent
  categories={Array []}
  getRecipesByCategory={[Function]}
/>
`;
